File: src/services/chatService.js
import apiClient from "../api/apiClient";

export const askQuestion = async (question, sessionId = null) => {

    const response = await apiClient.post("/chat", {

        question,

        sessionId

    });

    return response.data;

};

export const getChatSessions = async () => {

    const response = await apiClient.get("/chat/sessions");

    return response.data;

};

export const getConversation = async (sessionId) => {

    const response = await apiClient.get(

        `/chat/sessions/${sessionId}`

    );

    return response.data;

};

export const renameSession = async (sessionId, title) => {

    const response = await apiClient.put(

        `/chat/sessions/${sessionId}`,

        {

            title

        }

    );

    return response.data;

};

export const deleteSession = async (sessionId) => {

    const response = await apiClient.delete(

        `/chat/sessions/${sessionId}`

    );

    return response.data;

};

export const togglePin = async (sessionId) => {

    const response = await apiClient.put(

        `/chat/sessions/${sessionId}/pin`

    );

    return response.data;

};

File: ChatService.java
package com.campusgpt.service;

import java.util.List;

import com.campusgpt.dto.request.ChatRequest;
import com.campusgpt.dto.response.ChatResponse;
import com.campusgpt.dto.response.ChatSessionResponse;

import com.campusgpt.dto.response.ChatMessageResponse;

public interface ChatService {

	ChatResponse askQuestion(ChatRequest request, Long userId);

	List<ChatSessionResponse> getMySessions(Long userId);

	List<ChatMessageResponse> getConversation(String sessionId, Long userId);

	void renameSession(String sessionId, Long userId, String title);

	void deleteSession(String sessionId, Long userId);
	
	void togglePin(String sessionId, Long userId);

}